src/php/Compiler/Emitter/OutputEmitter.php

```diff
@@ -18,57 +18,39 @@
 final class OutputEmitter implements OutputEmitterInterface
 {
     private bool $enableSourceMaps;
-    private SourceMapGenerator $sourceMapGenerator;
     private NodeEmitterFactory $nodeEmitterFactory;
     private MungeInterface $munge;
     private PrinterInterface $printer;
 
     private int $indentLevel = 0;
-    private int $generatedLines = 0;
-    private int $generatedColumns = 0;
-    private array $sourceMap = [];
+    private SourceMapState $sourceMapState;
 
     public function __construct(
         bool $enableSourceMaps,
-        SourceMapGenerator $sourceMapGenerator,
         NodeEmitterFactory $nodeEmitterFactory,
         MungeInterface $munge,
         PrinterInterface $printer
     ) {
         $this->enableSourceMaps = $enableSourceMaps;
-        $this->sourceMapGenerator = $sourceMapGenerator;
         $this->nodeEmitterFactory = $nodeEmitterFactory;
         $this->munge = $munge;
         $this->printer = $printer;
+        $this->sourceMapState = new SourceMapState();
```

Comment:
I would prefer to inject via the constructor the dependencies, so the Factory is the place that orchestrates the instantiations and their dependencies. I think it's a good convention to have in the project. What do you think?